### DISCUSSION TITLE
Nesting profilings

### DISCUSSION THREAD

**Discussion by glensc** [perftools/xhgui#380] [Ideas]
_This is perhaps a discussion for [collector part](https://github.com/perftools/php-profiler), but I'll discuss it here to have better visibility and may involve xhgui itself too._

I realized that all profiles we support, do not support nesting, but probably it's possible to emulate nesting by merging the profiling data or extracting a subset of data in userspace.

**Comment by markstory** [comment]
What would create a nested profile?

**Reply by glensc** [reply]
With given example:

```php

1. $profiler->start();
2.   $profiler->start();
3.   $profiler->stop();
4. $profiler->stop();
```

in my code, I have overall profiler running (1), now I want to profile subset of code (2-3), but that should not interfere with the overall profiling results (4).
